packages/strapi-admin/domain/permission/index.js

```diff
@@ -0,0 +1,160 @@
+  return set('conditions', remove(eq(condition), permission.conditions), permission);
+});
+
+/**
+ * Gets a property or a part of a property from a permission.
+ * @param {string} property - The property to get
+ * @param {Permission} permission - The permission on which we want to access the property
+ * @return {Permission}
+ */
+const getProperty = curry((property, permission) => get(`properties.${property}`, permission));
+
+/**
+ * Set a value for a given property on a new permission object
+ * @param {string} property - The name of the property
+ * @param {any} value - The value of the property
+ * @param {Permission} permission - The permission on which we want to set the property
+ * @return {Permission}
+ */
+const setProperty = (property, value, permission) => {
+  return set(`properties.${property}`, value, permission);
+};
+
+/**
+ * Returns a new permission without the given property name set
+ * @param {string} property - The name of the property to delete
+ * @param {Permission} permission - The permission on which we want to remove the property
+ * @return {Permission}
+ */
+const deleteProperty = (property, permission) => omit(`properties.${property}`, permission);
+
+/**
+ * Creates a new {@link Permission} object from raw attributes. Set default values for certain fields
+ * @param {Permission} attributes
+ * @return {Permission}
+ */
+const create = attributes => {
+  return pipe(pick(permissionFields), merge(getDefaultPermission()))(attributes);
+};
+
+/**
+ * Using the given condition provider, check and remove invalid condition from the permission's condition array.
+ * @param {object} provider - The condition provider used to do the checks
+ * @param {Permission} permission - The condition to sanitize
+ * @return {Permission}
+ */
+const sanitizeConditions = curry((provider, permission) => {
+  if (!isArray(permission.conditions)) {
+    return permission;
+  }
+
+  return permission.conditions
+    .filter(condition => !provider.has(condition))
+    .reduce((perm, condition) => removeCondition(condition, perm), permission);
+});
+
+/**
+ * Transform raw attributes into valid permissions using the create domain function.
+ * @param {object | object[]} payload - Can either be a single object of attributes or an array of those objects.
+ * @return {Permission | Permission[]}
+ */
+const toPermission = payload => (isArray(payload) ? map(create, payload) : create(payload));
+
+/**
+ * Create a new abstract permission domain with a bound immutable permission and using the methods exposed by the given abstract domain factory
+ * @param {function(Permission):object} abstractDomainFactory - A factory that takes the initial permission as only parameter and can mutate it
+ * @param {Permission} permission - The permission that will be bound to the domain. It can only be mutated from the factory
+ * @return {{ permission: Permission} & object}
+ */
+const createBoundAbstractDomain = (abstractDomainFactory, permission) => {
```

Comment:
I don't think it make sens to have this in the domain.

Reply:
On which layer would you put this instead? The permission service? A private method next to where it'll be used?

Reply:
I would create the modifier directly in the service that needs it as it is still not reused anywhere let's not over complicate it